python/lsst/pipe/base/pipeline.py

```diff
@@ -747,6 +756,18 @@ def makeDatasetTypesSet(connectionType: str, freeze: bool = True) -> NamedValueS
                 datasetTypes.freeze()
             return datasetTypes
 
+        # optionally add initOutput dataset for config
+        initOutputs = makeDatasetTypesSet("initOutputs", freeze=False)
+        if include_configs:
```

Comment:
Should this logic be here or inside the makeDatasetTypeSet method? Do we only expect this to be used here?

Reply:
It's only appropriate for `initOutputs`, so I don't think we want it in `makeDatasetType`, since that's called for all of the connection types.